.eslintrc.json

```diff
@@ -412,6 +412,7 @@
                     "**/{vs,sql}/editor/{common,browser}/**",
                     "**/{vs,sql}/editor/contrib/**", // editor/contrib is equivalent to /browser/ by convention
                     "**/{vs,sql}/workbench/workbench.web.api",
+                    "**/{vs,sql}/workbench/api/**",
```

Comment:
Those types are meant to be only used on the exthost code. which is why this import rule doesn't apply. You should duplicate the type in the core code somewhere rather than just allowing this import rule.

Reply:
Why are contribution files allowed to use them then?

https://github.com/microsoft/azuredatastudio/blob/master/src/sql/workbench/contrib/scripting/browser/scripting.contribution.ts#L18

Reply:
@anthonydresser I've updated the imports, could you take a look?